### PR TITLE
Added new types to support translation from xml spec of slicer CLI

### PR DESCRIPTION
**Pull request by cdeepakroy**
Added new types -- integer, integer_list, number_list, string_list - to support translation from [xml spec of slicer execution model CLI](https://www.slicer.org/slicerWiki/index.php/Slicer3:Execution_Model_Documentation#XML_Schema)

### CODE REVIEW

File: girder_worker/format/integer/validate_json.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "inputs": [{"name": "input", "type": "integer", "format": "json"}],
+    "outputs": [{"name": "output", "type": "boolean", "format": "boolean"}],
+    "script": "output = isinstance(input, (str, unicode))",
+    "extensions": ["integer-json"],
```
Comment (zachmullen) [inline girder_worker/format/integer/validate_json.json]: What does extensions do?

**Comment by zachmullen** [comment]
I'm fine with these new types, but I'd like @jeffbaumes to weigh in on whether this is in line with how he thinks the type/format system should look.

File: girder_worker/format/integer/validate_json.json (hunk 1 above)
Comment (danlamanna) [inline girder_worker/format/integer/validate_json.json]: `extensions` are just a way for Flow to infer what type/format something is. If there are conflicts (i.e. tables and graphs could both theoretically be stored as a csv) then Flow forces the user to resolve the conflict.

File: tests/integer_list_test.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                    "format": "integer_list"
+                }
+            ],
+            "outputs": [{"name": "c",
+                         "type": "integer_list", "format": "integer_list"}],
+            "script": "c = a + b",
+            "mode": "python"
+        }
+
+    def test_integer(self):
```
Comment (jcfr) [inline tests/integer_list_test.py]: should this be named `test_interger_list` ?

**Comment by jcfr** [comment]
:+1:

File: tests/integer_list_test.py (hunk 2 above)
Comment (cdeepakroy) [inline tests/integer_list_test.py]: I just followed the naming convention used for the [rest of the tests](https://github.com/girder/girder_worker/tree/master/tests) in girder_worker.

**Comment by jeffbaumes** [comment]
I think this follows everything appropriately. This goes even further down the path of a flat list of types without complex typing logic, hence the explicit code for `number_list` instead of having some girder_worker `listof` construct to generally make a list type via something like `listof(number)`. I prefer staying on this simpler path - otherwise we will end up needing to use (or worse, create) an advanced typing system like [datashape](http://datashape.readthedocs.org/en/latest/) which would be heavyweight and complex - which is against what we are trying to do here I think.

All that to say LGTM.

**Comment by cdeepakroy** [comment]
Thanks for the input everyone! Can someone merge it? I don't seem to have rights.

**Comment by jcfr** [comment]
Et voila. 

(Someone with admin access would have to grant push access.)

**Comment by zachmullen** [comment]
@cdeepakroy I just invited you to the developers team, which should give you write access to all repos in this org.

**Pull request merged**